From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Wed, 27 Feb 2019 20:11:35 +0100
Subject: osk-layouts: Fix French layout

The script to convert XML keyboard layouts to json has a small bug
that causes the French-Canadian layout to end up as French.

Bug: https://gitlab.gnome.org/GNOME/gnome-shell/issues/997
Bug-Debian: https://bugs.debian.org/926452
Origin: upstream, 3.31.92, commit:859aef78c4d2472b2545ce9ecc889c00b9893494
---
 data/gnome-shell-osk-layouts.gresource.xml |   1 +
 data/osk-layouts/ca.json                   | 599 +++++++++++++++++++++++++++++
 data/osk-layouts/fr.json                   |  89 +++--
 3 files changed, 652 insertions(+), 37 deletions(-)
 create mode 100644 data/osk-layouts/ca.json

diff --git a/data/gnome-shell-osk-layouts.gresource.xml b/data/gnome-shell-osk-layouts.gresource.xml
index b7969c7..6c98a64 100644
--- a/data/gnome-shell-osk-layouts.gresource.xml
+++ b/data/gnome-shell-osk-layouts.gresource.xml
@@ -6,6 +6,7 @@
     <file>be.json</file>
     <file>bg.json</file>
     <file>by.json</file>
+    <file>ca.json</file>
     <file>cz.json</file>
     <file>de.json</file>
     <file>dk.json</file>
diff --git a/data/osk-layouts/ca.json b/data/osk-layouts/ca.json
new file mode 100644
index 0000000..2b2bedc
--- /dev/null
+++ b/data/osk-layouts/ca.json
@@ -0,0 +1,599 @@
+{
+  "levels": [
+    {
+      "level": "",
+      "mode": "default",
+      "rows": [
+        [
+          [
+            "q"
+          ],
+          [
+            "w"
+          ],
+          [
+            "e",
+            "é",
+            "è",
+            "ê",
+            "ë",
+            "%",
+            "ę",
+            "ė",
+            "ē"
+          ],
+          [
+            "r"
+          ],
+          [
+            "t"
+          ],
+          [
+            "y",
+            "%",
+            "ÿ"
+          ],
+          [
+            "u",
+            "ù",
+            "û",
+            "%",
+            "ü",
+            "ú",
+            "ū"
+          ],
+          [
+            "i",
+            "î",
+            "%",
+            "ï",
+            "ì",
+            "í",
+            "į",
+            "ī"
+          ],
+          [
+            "o",
+            "ô",
+            "œ",
+            "%",
+            "ö",
+            "ò",
+            "ó",
+            "õ",
+            "ø",
+            "ō",
+            "º"
+          ],
+          [
+            "p"
+          ]
+        ],
+        [
+          [
+            "a",
+            "à",
+            "â",
+            "%",
+            "æ",
+            "á",
+            "ä",
+            "ã",
+            "å",
+            "ā",
+            "ª"
+          ],
+          [
+            "s"
+          ],
+          [
+            "d"
+          ],
+          [
+            "f"
+          ],
+          [
+            "g"
+          ],
+          [
+            "h"
+          ],
+          [
+            "j"
+          ],
+          [
+            "k"
+          ],
+          [
+            "l"
+          ]
+        ],
+        [
+          [
+            "z"
+          ],
+          [
+            "x"
+          ],
+          [
+            "c",
+            "ç",
+            "ć",
+            "č"
+          ],
+          [
+            "v"
+          ],
+          [
+            "b"
+          ],
+          [
+            "n"
+          ],
+          [
+            "m"
+          ]
+        ],
+        [
+          [
+            ","
+          ],
+          [
+            " "
+          ],
+          [
+            ".",
+            "#",
+            "!",
+            ",",
+            "?",
+            "-",
+            ":",
+            "'",
+            "@"
+          ]
+        ]
+      ]
+    },
+    {
+      "level": "shift",
+      "mode": "latched",
+      "rows": [
+        [
+          [
+            "Q"
+          ],
+          [
+            "W"
+          ],
+          [
+            "E",
+            "É",
+            "È",
+            "Ê",
+            "Ë",
+            "%",
+            "Ę",
+            "Ė",
+            "Ē"
+          ],
+          [
+            "R"
+          ],
+          [
+            "T"
+          ],
+          [
+            "Y",
+            "%",
+            "Ÿ"
+          ],
+          [
+            "U",
+            "Ù",
+            "Û",
+            "%",
+            "Ü",
+            "Ú",
+            "Ū"
+          ],
+          [
+            "I",
+            "Î",
+            "%",
+            "Ï",
+            "Ì",
+            "Í",
+            "Į",
+            "Ī"
+          ],
+          [
+            "O",
+            "Ô",
+            "Œ",
+            "%",
+            "Ö",
+            "Ò",
+            "Ó",
+            "Õ",
+            "Ø",
+            "Ō",
+            "º"
+          ],
+          [
+            "P"
+          ]
+        ],
+        [
+          [
+            "A",
+            "À",
+            "Â",
+            "%",
+            "Æ",
+            "Á",
+            "Ä",
+            "Ã",
+            "Å",
+            "Ā",
+            "ª"
+          ],
+          [
+            "S"
+          ],
+          [
+            "D"
+          ],
+          [
+            "F"
+          ],
+          [
+            "G"
+          ],
+          [
+            "H"
+          ],
+          [
+            "J"
+          ],
+          [
+            "K"
+          ],
+          [
+            "L"
+          ]
+        ],
+        [
+          [
+            "Z"
+          ],
+          [
+            "X"
+          ],
+          [
+            "C",
+            "Ç",
+            "Ć",
+            "Č"
+          ],
+          [
+            "V"
+          ],
+          [
+            "B"
+          ],
+          [
+            "N"
+          ],
+          [
+            "M"
+          ]
+        ],
+        [
+          [
+            ","
+          ],
+          [
+            " "
+          ],
+          [
+            ".",
+            "#",
+            "!",
+            ",",
+            "?",
+            "-",
+            ":",
+            "'",
+            "@"
+          ]
+        ]
+      ]
+    },
+    {
+      "level": "opt",
+      "mode": "locked",
+      "rows": [
+        [
+          [
+            "1",
+            "¹",
+            "½",
+            "⅓",
+            "¼",
+            "⅛"
+          ],
+          [
+            "2",
+            "²",
+            "⅔"
+          ],
+          [
+            "3",
+            "³",
+            "¾",
+            "⅜"
+          ],
+          [
+            "4",
+            "⁴"
+          ],
+          [
+            "5",
+            "⅝"
+          ],
+          [
+            "6"
+          ],
+          [
+            "7",
+            "⅞"
+          ],
+          [
+            "8"
+          ],
+          [
+            "9"
+          ],
+          [
+            "0",
+            "ⁿ",
+            "∅"
+          ]
+        ],
+        [
+          [
+            "@"
+          ],
+          [
+            "#"
+          ],
+          [
+            "$",
+            "¢",
+            "£",
+            "€",
+            "¥",
+            "₱"
+          ],
+          [
+            "%",
+            "‰"
+          ],
+          [
+            "&"
+          ],
+          [
+            "-",
+            "_",
+            "–",
+            "—",
+            "·"
+          ],
+          [
+            "+",
+            "±"
+          ],
+          [
+            "(",
+            "<",
+            "{",
+            "["
+          ],
+          [
+            ")",
+            ">",
+            "}",
+            "]"
+          ]
+        ],
+        [
+          [
+            "*",
+            "†",
+            "‡",
+            "★"
+          ],
+          [
+            "\"",
+            "“",
+            "”",
+            "«",
+            "»"
+          ],
+          [
+            "'",
+            "‘",
+            "’",
+            "‹",
+            "›"
+          ],
+          [
+            ":"
+          ],
+          [
+            ";"
+          ],
+          [
+            "!",
+            "¡"
+          ],
+          [
+            "?",
+            "¿"
+          ]
+        ],
+        [
+          [
+            "_"
+          ],
+          [
+            "/"
+          ],
+          [
+            " "
+          ],
+          [
+            ","
+          ],
+          [
+            ".",
+            "…"
+          ]
+        ]
+      ]
+    },
+    {
+      "level": "opt+shift",
+      "mode": "locked",
+      "rows": [
+        [
+          [
+            "~"
+          ],
+          [
+            "`"
+          ],
+          [
+            "|"
+          ],
+          [
+            "•",
+            "♪",
+            "♥",
+            "♠",
+            "♦",
+            "♣"
+          ],
+          [
+            "√"
+          ],
+          [
+            "Π",
+            "π"
+          ],
+          [
+            "÷"
+          ],
+          [
+            "×"
+          ],
+          [
+            "¶",
+            "§"
+          ],
+          [
+            "∆"
+          ]
+        ],
+        [
+          [
+            "£"
+          ],
+          [
+            "¢"
+          ],
+          [
+            "€"
+          ],
+          [
+            "¥"
+          ],
+          [
+            "^",
+            "↑",
+            "↓",
+            "←",
+            "→"
+          ],
+          [
+            "°",
+            "′",
+            "″"
+          ],
+          [
+            "=",
+            "≠",
+            "≈",
+            "∞"
+          ],
+          [
+            "{"
+          ],
+          [
+            "}"
+          ]
+        ],
+        [
+          [
+            "\\"
+          ],
+          [
+            "©"
+          ],
+          [
+            "®"
+          ],
+          [
+            "™"
+          ],
+          [
+            "℅"
+          ],
+          [
+            "["
+          ],
+          [
+            "]"
+          ]
+        ],
+        [
+          [
+            "<",
+            "‹",
+            "≤",
+            "«"
+          ],
+          [
+            ">",
+            "›",
+            "≥",
+            "»"
+          ],
+          [
+            " "
+          ],
+          [
+            ","
+          ],
+          [
+            ".",
+            "…"
+          ]
+        ]
+      ]
+    }
+  ],
+  "locale": "fr-CA",
+  "name": "French Canada"
+}
\ No newline at end of file
diff --git a/data/osk-layouts/fr.json b/data/osk-layouts/fr.json
index 2b2bedc..dd572b1 100644
--- a/data/osk-layouts/fr.json
+++ b/data/osk-layouts/fr.json
@@ -6,10 +6,20 @@
       "rows": [
         [
           [
-            "q"
+            "a",
+            "à",
+            "â",
+            "%",
+            "æ",
+            "á",
+            "ä",
+            "ã",
+            "å",
+            "ā",
+            "ª"
           ],
           [
-            "w"
+            "z"
           ],
           [
             "e",
@@ -71,17 +81,7 @@
         ],
         [
           [
-            "a",
-            "à",
-            "â",
-            "%",
-            "æ",
-            "á",
-            "ä",
-            "ã",
-            "å",
-            "ā",
-            "ª"
+            "q"
           ],
           [
             "s"
@@ -106,11 +106,14 @@
           ],
           [
             "l"
+          ],
+          [
+            "m"
           ]
         ],
         [
           [
-            "z"
+            "w"
           ],
           [
             "x"
@@ -131,7 +134,11 @@
             "n"
           ],
           [
-            "m"
+            "'",
+            "‘",
+            "’",
+            "‹",
+            "›"
           ]
         ],
         [
@@ -161,10 +168,20 @@
       "rows": [
         [
           [
-            "Q"
+            "A",
+            "À",
+            "Â",
+            "%",
+            "Æ",
+            "Á",
+            "Ä",
+            "Ã",
+            "Å",
+            "Ā",
+            "ª"
           ],
           [
-            "W"
+            "Z"
           ],
           [
             "E",
@@ -226,17 +243,7 @@
         ],
         [
           [
-            "A",
-            "À",
-            "Â",
-            "%",
-            "Æ",
-            "Á",
-            "Ä",
-            "Ã",
-            "Å",
-            "Ā",
-            "ª"
+            "Q"
           ],
           [
             "S"
@@ -261,11 +268,14 @@
           ],
           [
             "L"
+          ],
+          [
+            "M"
           ]
         ],
         [
           [
-            "Z"
+            "W"
           ],
           [
             "X"
@@ -286,7 +296,11 @@
             "N"
           ],
           [
-            "M"
+            "'",
+            "‘",
+            "’",
+            "‹",
+            "›"
           ]
         ],
         [
@@ -369,10 +383,10 @@
             "#"
           ],
           [
-            "$",
+            "€",
             "¢",
             "£",
-            "€",
+            "$",
             "¥",
             "₱"
           ],
@@ -511,13 +525,14 @@
             "£"
           ],
           [
-            "¢"
+            "¥"
           ],
           [
-            "€"
+            "$",
+            "¢"
           ],
           [
-            "¥"
+            "¢"
           ],
           [
             "^",
@@ -594,6 +609,6 @@
       ]
     }
   ],
-  "locale": "fr-CA",
-  "name": "French Canada"
+  "locale": "fr",
+  "name": "French"
 }
\ No newline at end of file
